### PR TITLE
Tile server: ensure timezone-naive time selector

### LINKED ISSUE #555: Address `xcube serve`  FutureWarning: timezone-naive DatetimeIndex with a timezone-aware datetime is deprecated
When serving xcube datasets with `xcube serve` there is a 
`FutureWarning: Indexing a timezone-naive DatetimeIndex with a timezone-aware datetime is deprecated and will raise KeyError in a future version.  Use a timezone-naive object instead.`
We need to address this before we run into that KeyError.

### PR DESCRIPTION
**Pull request by pont-us**
Fixes #555.

In tile._get_var_2d_array (used by new_color_mapped_image),
check for the possibility that the array uses datetime64 and the
indexer is timezone-aware. In this case, convert to a timezone-naive
indexer before performing the selection.

[Description of PR]

Checklist:

* ~~[ ] Add unit tests and/or doctests in docstrings~~ (not needed)
* ~~[ ] Add docstrings and API docs for any new/modified user-facing classes and functions~~ N/A
* ~~[ ] New/modified features documented in `docs/source/*`~~ N/A
* ~~[ ] Changes documented in `CHANGES.md`~~ N/A
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

Remember to close associated issues after merge!

### CODE REVIEW

**Review by AliceBalfanz — APPROVED** [review]
Tested locally and works well :) Thanks for fixing!

**Review by forman — CHANGES_REQUESTED** [review]
- The natural home of this fix should be `xcube.core.tile.parse_non_spatial_labels()`
- I assume, the nature of the problem requires a more general solution. Hence, we need an understanding of the root cause and a general function that fixes the issue before timstamps are passed into `xarray.Dataset/DataArray.sel()`.

File: xcube/core/tile.py [hunk 1]
```diff
@@ -242,14 +242,34 @@ def _get_var_2d_array(var: xr.DataArray,
...
+                            labels: Dict[str, Any]) -> Dict[str, Any]:
+    """Ensure that labels['time'] is timezone-naive, if necessary.
+    If var has a 'time' dimension of type datetime64 and labels has a 'time'
+    key with a timezone-aware value, return a modified labels dictionary with
+    a timezone-naive time value. Otherwise return the original labels.
+    """
+    if 'time' in var.dims and \
+        hasattr(var['time'], 'dtype') and \
+        hasattr(var['time'].dtype, 'type') and \
+        var['time'].dtype.type is np.datetime64 and \
```
Comment (TonioF) [inline xcube/core/tile.py:264]: For readability, I'd prefer if this was moved into a helper function

**Review by TonioF — COMMENTED** [review]
I really would like a test for this. Also, I'd expect this issue to arise in more places than this. Did you check the other places?

File: xcube/core/tile.py (hunk 1 above)
Comment (pont-us) [reply xcube/core/tile.py]: Seems sensible -- done.

**Comment by pont-us** [comment]
> I really would like a test for this.

OK, added.

**Comment by pont-us** [comment]
> Did you check the other places?

There seem to be fewer of them than I feared. I found time selection used in three other places in the xcube codebase:

 - `core.select.select_temporal_subset`
 - `core.timeseries.get_time_series`
 - `cli.tile.tile`

The last of these will be made safe by Norman's suggestion of moving the check to `core.tile.parse_non_spatial_labels()`, but the other two will require further work.

**Comment by pont-us** [comment]
> The natural home of this fix should be `xcube.core.tile.parse_non_spatial_labels()`

OK, moved.

**Comment by forman** [comment]
> but the other two will require further work.

Then let's just fix the server and enhance the issue #555 by your findings or create a new one.

**Pull request merged**